src/AspNetCoreModuleV2/InProcessRequestHandler/dllmain.cxx

```diff
@@ -103,31 +111,47 @@ __stdcall
 CreateApplication(
     _In_  IHttpServer        *pServer,
     _In_  IHttpContext       *pHttpContext,
-    _In_  PCWSTR             pwzExeLocation,
+    _In_  PCWSTR              pwzExeLocation,
     _Out_ IAPPLICATION      **ppApplication
 )
 {
-    HRESULT      hr = S_OK;
-    IAPPLICATION *pApplication = NULL;
-    REQUESTHANDLER_CONFIG *pConfig;
+    HRESULT                 hr = S_OK;
+    IN_PROCESS_APPLICATION *pApplication = NULL;
+    REQUESTHANDLER_CONFIG  *pConfig;
+
     // Initialze some global variables here
     InitializeGlobalConfiguration(pServer);
 
     hr = REQUESTHANDLER_CONFIG::CreateRequestHandlerConfig(pServer, pHttpContext->GetApplication(), &pConfig);
     if (FAILED(hr))
     {
-        return hr;
+        goto Finished;
     }
 
-    pApplication = new IN_PROCESS_APPLICATION(pServer, pConfig, pwzExeLocation);
-    if (pApplication == NULL)
+    pApplication = new IN_PROCESS_APPLICATION(pServer, pConfig);
+
```

Comment:
you should do NULL check for pApplication

Reply:
It can't be NULL as new will never return NULL.

Reply:
just checked IISOOB and IIS, it seems we always link to nothrownew.obj to avoid exception

Reply:
I disagree. It has been the standard to throw on new for the last 15 years. If we are out of memory, I would like to crash the process (or we can catch it).

Reply:
we need at least add nothrownew for out process and shim

Reply:
I'll make a separate issue for it.

Reply:
I would argue that we don't need nothrownew in shim

Reply:
Thank about there was a bug in our code which leads to null ref under stress. In old patten,  worker process will crash, while with try-catch, client occasionally receive 500 response. Will it be easy to debug the root cause for the latter?

Reply:
Sorry, I didn't really understand. I don't think there is such a thing as null reference exception in C++ so behavior in this case wouldn't change at all.

Reply:
Yeah, we would probably AV or have undefined behavior.

Reply:
if new failed here, exception will be thrown and pConfig will be leaked.